Snitch does not report HTTP errors properly

### Steps to reproduce

- Replace all occurances of `api.github.com` with `localhost`
- Recompile snitch
- Try to snitch up something

### Observed

```
Detected GitHub project: https://github.com/tsoding/snitch-lab
Main.java:6:         // TODO: test
Do you want to report this?  [y/n] y
panic: interface conversion: interface {} is nil, not float64

goroutine 1 [running]:
main.Todo.ReportTodo(0xc4203d2340, 0xb, 0xc4203d2351, 0x4, 0xc420012970, 0x4, 0x0, 0xc4202ba280, 0x9, 0x6, ...)
	/home/rexim/Programming/tsoding/snitch/todo.go:185 +0x41c
main.reportSubcommand(0xc4200e84e0, 0xc420070440, 0x4, 0x4, 0xc4200182d1, 0x28, 0xc4200166a0, 0x12, 0x0, 0x0, ...)
	/home/rexim/Programming/tsoding/snitch/main.go:66 +0x313
main.main()
	/home/rexim/Programming/tsoding/snitch/main.go:301 +0x76e
```

### Expected

Some kind of indication that connection was not estableshed.

--- a/github_credentials.go
+++ b/github_credentials.go
@@ -3,9 +3,9 @@ package main
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"gopkg.in/go-ini/ini.v1"
 	"net/http"
-	"fmt"
 )
 
 // GithubCredentials contains PersonalToken for GitHub API authorization

--- a/github_credentials.go
+++ b/github_credentials.go
@@ -5,6 +5,7 @@ import (
 	"encoding/json"
 	"gopkg.in/go-ini/ini.v1"
 	"net/http"
+	"fmt"
 )
 
 // GithubCredentials contains PersonalToken for GitHub API authorization
@@ -33,6 +34,12 @@ func (creds GithubCredentials) QueryGithubAPI(method, url string, jsonBody map[s
 		return nil, err
 	}
 	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(resp.Body)
+		return nil, fmt.Errorf("GitHub API error: %s", buf.String())
+	}
 
 	var v map[string]interface{}
 	if err := json.NewDecoder(resp.Body).Decode(&v); err != nil {

--- a/todo.go
+++ b/todo.go
@@ -181,6 +181,9 @@ func (todo Todo) ReportTodo(creds GithubCredentials, repo string, body string) (
 			"title": todo.Title,
 			"body":  body,
 		})
+	if err != nil {
+		return todo, err
+	}
 
 	id := "#" + strconv.Itoa(int(json["number"].(float64)))
 	todo.ID = &id

--- a/todo.go
+++ b/todo.go
@@ -173,7 +173,6 @@ func (todo Todo) RetrieveGithubStatus(creds GithubCredentials, repo string) (str
 // ReportTodo reports the todo as a Github Issue, updates the file
 // where the todo is located and commits the changes to the git repo.
 func (todo Todo) ReportTodo(creds GithubCredentials, repo string, body string) (Todo, error) {
-	// TODO(#60): ReportTodo is not a Todo method
 	json, err := creds.QueryGithubAPI(
 		"POST",
 		"https://api.github.com/repos/"+repo+"/issues",

